www/source/docs/using-builder.html.md.erb

```diff
@@ -236,59 +360,248 @@ Initialize the setting from the command line with:
+
+Origin secrets are retained by the origin and are available for any of that origin's packages. The origin secrets in your local environment are encrypted with an origin encryption key. Only Chef Habitat Builder can read encrypted origin secrets.
+
+#### Manage Origin Secrets with the Chef Habitat CLI
+
+You can view the list of origin secrets and delete them in Chef Habitat Builder.
+However, the primary way of interacting with origin secrets is with the Chef Habitat CLI.
+
+Prerequisites:
+
+* [Download the Chef Habitat CLI](/docs/install-habitat)
+* [Create a Chef Habitat Builder account](/#builder-account)
+* [Generate a personal access token](/#builder-token)
+* [Create an origin](/#create-origin)
+* [Get origin keys](/#origin-keys)
+
+##### List Secrets
+
+To list all of the secrets in an origin, use:
+
+```hab
+hab origin secret list --origin <ORIGIN>
 ```
 
-You can also set your origin and token as environment variables:
+##### Set Origin Secrets as Environment Variables
+
+Add your origin secrets as environment variables in your local environment:
 
 ```bash
 export HAB_ORIGIN=<ORIGIN>
 export HAB_AUTH_TOKEN=<TOKEN>
 hab origin secret list
 ```
 
-To create a new secret, give the secret a name and string value:
+##### Save an Origin Secret
+
+To save an origin secret give the secret a name and the key value:
+
+```hab
+hab origin secret upload AWS_ACCESS_KEY_ID <your-key-id>
+hab origin secret upload AWS_SECRET_ACCESS_KEY <your-secret-access-key>
+```
+
+The output should similar to:
+
+```bash
+$ hab origin secret upload AWS_ACCESS_KEY_ID 1234567890EXAMPLE
+↓ Downloading latest public encryption key
+    79 B / 79 B | [========================================] 100.00 % 120.23 KB/s
+☑ Cached habicat-20200123456789.pub
+☛ Encrypting value for key AWS_ACCESS_KEY_ID.
+✓ Encrypted AWS_ACCESS_KEY_ID=[REDACTED].
+↑ Uploading secret for key AWS_ACCESS_KEY_ID.
+✓ Uploaded secret for AWS_ACCESS_KEY_ID.
+```
+
+##### Delete an Origin Secret
+
+To delete an origin secret from an origin with the CLI
+
+```hab
+hab origin secret delete AWS_ACCESS_KEY_ID
+hab origin secret delete AWS_SECRET_ACCESS_KEY
+```
+
+See [Using Origin Secrets in Plans](/docs/developing-packages/#buildtime-workflow) for guidance on using origin secrets.
+
+See the [`hab origin secret`](/docs/habitat-cli/#hab-origin-secret) CLI documentation for more information on these commands.
+
+---
+## <a name="origin-members" id="origin-members" data-magellan-target="origin-members">Origin Membership</a>
+
+Prerequisites:
+
+* [Download the Chef Habitat CLI](/docs/install-habitat)
+* [Create a Chef Habitat Builder account](/#builder-account)
+* [Generate a personal access token](/#builder-token)
+* [Create an origin](/#create-origin)
+* [Get origin keys](/#origin-keys)
+
+### Role-Based Access Control (RBAC) for Chef Habitat Builder (SaaS and on-prem)
+
+TODO:New in: SEMVER
```

Comment:
Is this intentional?

Reply:
Yes, I need to replace it (and write a Buildkite lint that looks for my todos, but only in docs files)

Reply:
Fixed.